spec/support/change_streams.rb

```diff
@@ -165,8 +177,12 @@ def match_result?(result)
         end
 
         def match_commands?(actual)
-          @expectations.each_with_index.all? do |e, i|
-            actual[i] && match?(e, actual[i])
+          if @expectations
```

Comment:
I would prefer this condition be given in the spec test rather than in the matcher.

Example 1: https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/spec_tests/uri_options_spec.rb#L54

Example 2: https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/support/crud.rb#L153 (the crud tests have a bit more than usual meta programming around them)